File: code/go/ch18/src/luago/api/lua_auxlib.go
package api

type FuncReg map[string]GoFunction

// auxiliary library
type AuxLib interface {
	/* Error-report functions */
	Error2(fmt string, a ...interface{}) int
	ArgError(arg int, extraMsg string) int
	/* Argument check functions */
	CheckStack2(sz int, msg string)
	ArgCheck(cond bool, arg int, extraMsg string)
	CheckAny(arg int)
	CheckType(arg int, t LuaType)
	CheckInteger(arg int) int64
	CheckNumber(arg int) float64
	CheckString(arg int) string
	OptInteger(arg int, d int64) int64
	OptNumber(arg int, d float64) float64
	OptString(arg int, d string) string
	/* Load functions */
	DoFile(filename string) bool
	DoString(str string) bool
	LoadFile(filename string) int
	LoadFileX(filename, mode string) int
	LoadString(s string) int
	/* Other functions */
	TypeName2(idx int) string
	ToString2(idx int) string
	Len2(idx int) int64
	GetSubTable(idx int, fname string) bool
	GetMetafield(obj int, e string) LuaType
	CallMeta(obj int, e string) bool
	OpenLibs()
	RequireF(modname string, openf GoFunction, glb bool)
	NewLib(l FuncReg)
	NewLibTable(l FuncReg)
	SetFuncs(l FuncReg, nup int)
}

File: code/java/ch18/src/main/java/com/github/zxh0/luago/api/LuaAuxLib.java
package com.github.zxh0.luago.api;

import java.util.Map;

public interface LuaAuxLib {

    //type FuncReg map[string]GoFunction

    /* Error-report functions */
    int error2(String fmt, Object... a);
    int argError(int arg, String extraMsg);
    /* Argument check functions */
    void checkStack2(int sz, String msg);
    void argCheck(boolean cond, int arg, String extraMsg);
    void checkAny(int arg);
    void checkType(int arg, LuaType t);
    long checkInteger(int arg);
    double checkNumber(int arg);
    String checkString(int arg);
    long optInteger(int arg, long d);
    double optNumber(int arg, double d);
    String optString(int arg, String d);
    /* Load functions */
    boolean doFile(String filename);
    boolean doString(String str);
    ThreadStatus loadFile(String filename);
    ThreadStatus loadFileX(String filename, String mode);
    ThreadStatus loadString(String s);
    /* Other functions */
    String typeName2(int idx);
    String toString2(int idx);
    long len2(int idx);
    boolean getSubTable(int idx, String fname);
    LuaType getMetafield(int obj, String e);
    boolean callMeta(int obj, String e);
    void openLibs();
    void requireF(String modname, JavaFunction openf, boolean glb);
    void newLib(Map<String, JavaFunction> l);
    void newLibTable(Map<String, JavaFunction> l);
    void setFuncs(Map<String, JavaFunction> l, int nup);

}
